### PR TITLE
Various fixes for the DocUploadTask

### LINKED ISSUE #1458: Some documentation task do not build preview URLs
Example on this revision https://phabricator.services.mozilla.com/D163141

Screenshot 2022-12-07 at 10-51-16 ⚙ D163141 Bug 1792238 - Create a moz-button-group component r hjones

### LINKED ISSUE #1459: Enhance documentation files detection in DocUploadTask
As discussed in https://github.com/mozilla/code-review/issues/1458#issuecomment-1340812790, we should only keep documentation files with the extensions `.rst` or `.md`.

### PR DESCRIPTION
**Pull request by EvaBardou**
Closes #1459    
Closes #1458

### CODE REVIEW

File: bot/code_review_bot/tasks/docupload.py [hunk 1]
```diff
@@ -77,8 +77,18 @@ def build_notice(self, artifacts, revision):
...
+            file
+            for file in revision.files
+            if "docs" in file and file.endswith((".rst", ".md"))
+        ]
         nb_docs = len(doc_files)
+        if not nb_docs:
+            logger.info(
+                "Found no documentation file in revision, skipping comment creation"
+            )
+            return ""
```
Comment (La0) [inline bot/code_review_bot/tasks/docupload.py:90]: You should just use `return` here (I mean return `None`) and the calling code should behave accordingly, skipping publication

Comment (EvaBardou) [reply bot/code_review_bot/tasks/docupload.py:90]: `""` is returned [above](https://github.com/mozilla/code-review/pull/1460/files#diff-774da06547f30fc28ca6c29ce9a4c274e67c173fbad29f840dd69087644cec3dR72), we aren't completely skipping the comment creation (I guess the log is a little misleading), we just don't append anything, related to the documentation, at the end of the existing one

File: bot/tests/test_docupload_task.py [hunk 2]
```diff
@@ -34,7 +34,7 @@
 
 - file [docs/folderB/folderC/index.rst](http://firefox-test-docs.mozilla.org/section2/index.html)
 
-- file [docs/folderB/folderC/subfolder/index.rst](http://firefox-test-docs.mozilla.org/section2/subfolder/index.html)
+- file [docs/folderB/folderC/subfolder/index.md](http://firefox-test-docs.mozilla.org/section2/subfolder/index.html)
```
Comment (marco-c) [inline bot/tests/test_docupload_task.py:37]: Nit: why changing from .rst to .md?

Comment (EvaBardou) [reply bot/tests/test_docupload_task.py:37]: It's just to assert that we properly handle `.md` files too, there wasn't one in the test before.

File: bot/code_review_bot/tasks/docupload.py [hunk 3]
```diff
@@ -77,8 +77,18 @@ def build_notice(self, artifacts, revision):
         if trees is None:
             logger.warn("Missing trees.json")
 
-        doc_files = [file for file in revision.files if "docs" in file]
+        doc_files = [
+            file
+            for file in revision.files
+            if "docs" in file and file.endswith((".rst", ".md"))
```
Comment (marco-c) [inline bot/code_review_bot/tasks/docupload.py:83]: Could you add a comment mentioning that we should keep the list up to date with https://searchfox.org/mozilla-central/rev/2d24d893669ad0fe8d76b0427b25369d35fcc19b/docs/conf.py#79 ?

**Comment by sylvestre** [comment]
btw, I love this feature, many thanks :)

**Comment by EvaBardou** [comment]
You're welcome!

**Pull request merged**